float64 delay_in_step         # instantaneous delay per simulation step, this must be less than delayMaxPerStep.
float64 delay_in_window       # total delay in current window period.
float64 delay_window_remain   # time left in current window period, before next budget reset.
